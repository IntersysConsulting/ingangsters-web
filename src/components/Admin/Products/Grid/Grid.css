.row {
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  flex-wrap: wrap;
  margin: auto;
}

@media screen and (max-width: 550px) {
  .row {
    flex-direction: column;
  }
}

@media screen and (max-width: 380px) {
  .row {
    width: 90%;
  }
}
