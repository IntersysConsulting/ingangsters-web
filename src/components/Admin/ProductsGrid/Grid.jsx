import React from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Product from "./ProductCard/Card";
const ProductsGrid = () => {
  return (
    <Container>
      <Row>
        <Product
          productName="This product"
          productStock={10}
          productImage="https://img.icons8.com/cute-clipart/64/000000/box.png"
        />
      </Row>
    </Container>
  );
};

export default ProductsGrid;
