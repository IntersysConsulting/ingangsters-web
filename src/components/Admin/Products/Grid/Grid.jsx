import React from "react";
import Product from "../ProductCard/Card";
import "./Grid.css";
import { connect } from "react-redux";
import LoadingBall from "../LoadingBall/LoadingBall";
import { fetchProducts } from "../../../../actions/creators/adminProducts";

const ProductsGrid = ({ items, loading, finishedFetch, fetchProducts }) => {
  if (items.length === 0 && !loading && !finishedFetch) {
    fetchProducts(7);
  }
  if (loading) {
    return (
      <div className="row">
        <LoadingBall />
      </div>
    );
  } else {
    const cards = items.map(item => (
      <Product
        {...item}
        key={item._id}
        onClick={() => {
          console.log("Clicked " + item._id);
        }}
      />
    ));
    return <div className="row">{cards}</div>;
  }
};

const mapStateToProps = state => ({
  items: state.adminProducts.list,
  loading: state.adminProducts.fetching,
  finishedFetch: state.adminProducts.finishedFetching
});

export default connect(
  mapStateToProps,
  { fetchProducts }
)(ProductsGrid);
