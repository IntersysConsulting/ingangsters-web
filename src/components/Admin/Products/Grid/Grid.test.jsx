import React from "react";
import { shallow } from "enzyme";
import { ProductsGrid } from "./Grid";

describe("ADMIN: Products Grid", () => {
  it("renders loading animation at start", () => {
    const wrapper = shallow(
      <ProductsGrid
        items={[]}
        loading={false}
        finishedFetch={false}
        fetchProducts={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders some items after loading", () => {
    const mockData = [
      {
        _id: "5d7676f35db4e862cc84027d",
        available: true,
        image:
          "https://i.kinja-img.com/gawker-media/image/upload/s--m6F4tJ-n--/c_scale,f_auto,fl_progressive,q_80,w_800/imfz8sq2jjpl19apqhvb.jpg",
        name: "Nintendo Switch",
        onClick: "/admin/product/5d7676f35db4e862cc84027d",
        price: 600000,
        shippable: true,
        stock: 1
      }
    ];

    const wrapper = shallow(
      <ProductsGrid
        items={mockData}
        loading={false}
        finishedFetch={true}
        fetchProducts={() => {}}
      />
    );
    expect(wrapper.find("Card").length).not.toBe(0);
  });
});
