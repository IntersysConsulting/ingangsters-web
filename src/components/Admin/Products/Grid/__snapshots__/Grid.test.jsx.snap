// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADMIN: Products Grid renders loading animation at start 1`] = `
<div
  className="row adminProductGrid mt-4"
/>
`;
